src/coreComponents/physicsSolvers/SolverStatistics.hpp

```diff
@@ -22,176 +22,377 @@
    */
-  integer getNumTimeSteps() const
-  { return m_numTimeSteps; }
+  void setFilename( string_view filename )
+  { m_iterationsFilename = filename; }
 
-  /**
-   * @return Number of time step cuts
-   */
-  integer getNumTimeStepCuts() const
-  { return m_numTimeStepCuts; }
 
   /**
-   * @return Cumulative number of successful outer loop iterations
+   * @return  A const iteration filename
    */
-  integer getNumSuccessfulOuterLoopIterations() const
-  { return m_numSuccessfulOuterLoopIterations; }
+  string const & getFilename() const
+  { return m_iterationsFilename;  }
 
   /**
-   * @return Cumulative number of successful nonlinear iterations
+   * @brief  A string reference to an iteration filename
    */
-  integer getNumSuccessfulNonlinearIterations() const
-  { return m_numSuccessfulNonlinearIterations; }
+  string & getFilename()
+  { return m_iterationsFilename;  }
+
 
   /**
-   * @return Cumulative number of successful linear iterations
+   * @brief Set the filename output file.
+   * @param name The filename as a string_view.
    */
-  integer getNumSuccessfulLinearIterations() const
-  { return m_numSuccessfulLinearIterations; }
+  void setTableName( string_view name )
+  { m_tableIterationName = name; }
 
   /**
-   * @return Cumulative number of discarded outer loop iterations
+   * @brief Output the statistics to the console in table format
    */
-  integer getNumDiscardedOuterLoopIterations() const
-  { return m_numDiscardedOuterLoopIterations; }
+  void outputStatistics();
 
   /**
-   * @return Cumulative number of discarded nonlinear iterations
+   * @brief Close the stream output file
    */
-  integer getNumDiscardedNonlinearIterations() const
-  { return m_numDiscardedNonlinearIterations; }
+  void closeFile()
+  { m_logStream.close(); }
+
+private:
+  /// A boolean indicating whether the CSV file is open or not
+  bool m_isCSVOpen = false;
+  /// Stream output for the iteration statistics
+  std::ofstream m_logStream;
+  /// Table Layout contenaning header for both CSV and log
+  std::unique_ptr< TableLayout > m_iterationCSVLayout;
+  /// Contain the iteration data for both CSV and log output
+  TableData m_iterationData;
+  /// Format the iteration statistics for the CSV file
+  std::unique_ptr< TableCSVFormatter > m_iterationCSVFormatter;
+  /// String name for the log table interation
+  string m_tableIterationName;
+  /// Filename for the iteration CSV file.
+  string m_iterationsFilename;
+};
+
+/**
+ * @brief Class containing convergence information given a time-step
+ */
+class ConvergenceStatistics
+{
+public:
 
   /**
-   * @return Cumulative number of discarded linear iterations
+   * @brief Construct a new Convergence Statistics object
    */
-  integer getNumDiscardedLinearIterations() const
-  { return m_numDiscardedLinearIterations; }
+  ConvergenceStatistics();
+
+  /// State of csv output. True when writeSolver is set to 2
+  bool m_csvOutput = false;
+
```

Comment:
Maybe you want to split that in two:
```suggestion
  /// True when statistics output flags is set to 2
  bool m_csvOutputEnabled = false;

  /// Current state of csv output.
  bool m_csvOutputOpened = false;
```

And, as I requested private attributes, you can check for one flag or the other depending on if it is file or data related.